src/replicator.rs

```diff
@@ -348,7 +348,7 @@ pub type PropertyEncryptor = fn(
     algorithm: Option<String>,
     kid: Option<String>,
     error: &Error,
-) -> Vec<u8>;
+) -> std::result::Result<Vec<u8>, Box<dyn std::error::Error>>;
```

Comment:
Why Box<dyn std::error::Error>?

Reply:
In order to manage any kind of error users can have. Do you think about using the unit `()` type instead for errors ? 
I think about if we want to propagate it up to high level caller that may want to be aware of specific encryption error cases.